middleware/locale/method/cookie_test.go

```diff
@@ -0,0 +1,33 @@
+package method_test
+
+import (
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/mholt/caddy/middleware/locale/method"
+)
+
+func TestCookieParsing(t *testing.T) {
+	cookie := method.Names["cookie"]
+	settings := &method.Settings{CookieName: "locale"}
+
+	request, _ := http.NewRequest("GET", "/", nil)
+
+	tests := []struct {
+		name            string
+		value           string
+		expectedLocales []string
+	}{
+		{"locale", "en", []string{"en"}},
+		{"locale", "de", []string{"de"}},
+	}
+
+	for index, test := range tests {
+		request.Header.Set("Cookie", (&http.Cookie{Name: test.name, Value: test.value}).String())
+		locales := cookie(request, settings)
```

Comment:
So you're basically duplicating Golang's tests if setting and retrieving HTTP Cookies work. That's redundant.